fecgov/openFEC
# Add flake8 to pytest for automated pep8 testing

## Summary (required)

Resolves #4257 

- Automate pep8 checking with `pytest-flake8` (See "how to test" for instructions for use)
- Fix presidential tests
- Update pytest, speed up tests, fix deprecation errors/warning

max line length = 120 chars

**Skipping**
- E128 continuation line under-indented for visual indent
- E127 continuation line over-indented for visual indent
- W503 line break before binary operator


## How to test the changes locally

- check out this branch 
- `pip install -r requirements.txt`
- `pytest` includes `pytest-flake8` PEP8 linting
- `pytest --no-linting` skips `pytest-flake8` PEP8 linting
- `pytest --linting` only runs `pytest-flake8` PEP8 linting
- `pytest --cache-clear --linting` to run without cached results https://github.com/tholo/pytest-flake8/issues/62
- You can still run one tests with `pytest tests/test_elections.py --no-linting` or `--linting` or don't specify either
